core/components/scripts/scripts.tsx

```diff
@@ -0,0 +1,59 @@
+import { removeEdgesAndNodes } from '@bigcommerce/catalyst-client';
+import Script from 'next/script';
+import { ComponentProps } from 'react';
+
+import { FragmentOf } from '~/client/graphql';
+
+import { ScriptsFragment } from './fragment';
+
+type ScriptsData = FragmentOf<typeof ScriptsFragment>;
+
+interface ScriptRendererProps {
+  scripts: ScriptsData['headerScripts'] | null;
+  strategy: ComponentProps<typeof Script>['strategy'];
+}
+
+export const ScriptManagerScripts = ({ scripts, strategy }: ScriptRendererProps) => {
+  if (!scripts?.edges) return null;
+
+  const scriptNodes = removeEdgesAndNodes(scripts);
+
+  return (
+    <>
+      {scriptNodes.map((script) => {
+        const scriptProps: ComponentProps<typeof Script> = {
+          strategy,
+        };
+
+        // Handle external scripts (SrcScript)
+        if (script.__typename === 'SrcScript' && script.src) {
+          scriptProps.src = script.src;
+
+          // Add integrity hashes if provided
+          if (script.integrityHashes.length > 0) {
+            scriptProps.integrity = script.integrityHashes.map((hashObj) => hashObj.hash).join(' ');
+            scriptProps.crossOrigin = 'anonymous';
+          }
+        }
+
+        // Handle inline scripts (InlineScript)
+        if (script.__typename === 'InlineScript' && script.scriptTag) {
+          const scriptMatch = /<script[^>]*>([\s\S]*?)<\/script>/i.exec(script.scriptTag);
+
+          if (scriptMatch?.[1]) {
+            scriptProps.dangerouslySetInnerHTML = {
+              __html: scriptMatch[1],
+            };
+          }
+        }
```

Comment:
I think we should have `dom-purify` here in the repo that you might want to use instead of a regex 🤔